assets/backport/github/github.go

```diff
@@ -0,0 +1,311 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package github
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/gravitational/trace"
+
+	go_github "github.com/google/go-github/v41/github"
+	"golang.org/x/oauth2"
+)
+
+type Client struct {
+	Client *go_github.Client
+	Config
+}
+
+type Config struct {
+	Token        string
+	Organization string
+	Repository   string
+}
+
+// New returns a new GitHub client.
+func New(ctx context.Context, c *Config) (*Client, error) {
+	if err := c.Check(); err != nil {
+		return nil, trace.Wrap(err)
+	}
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: c.Token},
+	)
+	return &Client{
+		Client: go_github.NewClient(oauth2.NewClient(ctx, ts)),
+		Config: *c,
+	}, nil
+}
+
+// Check validates config.
+func (c *Config) Check() error {
+	if c.Token == "" {
+		return trace.BadParameter("missing parameter Token")
+	}
+	if c.Organization == "" {
+		return trace.BadParameter("missing parameter Organization")
+	}
+	if c.Repository == "" {
+		return trace.BadParameter("missing parameter Repository")
+	}
+	return nil
+}
+
+// Backport backports changes from backportBranchName to a new branch based
+// off baseBranchName.
+//
+// A new branch is created with the name in the format of
+// auto-backport/[pull number]-to-[base branch], and
+// cherry-picks commits onto the new branch.
+func (c *Client) Backport(ctx context.Context, baseBranchName string, pullNumber int) (string, error) {
+	newBranchName := fmt.Sprintf("auto-backport/%v-to-%s", pullNumber, baseBranchName)
+	// Create a new branch off of the target branch.
+	err := c.createBranchFrom(ctx, baseBranchName, newBranchName)
+	if err != nil {
+		return "", trace.Wrap(err)
+	}
+	fmt.Printf("Created a new branch: %s.\n", newBranchName)
+
+	commits, err := c.getPullRequestCommits(ctx, pullNumber)
+	if err != nil {
+		return "", trace.Wrap(err)
+	}
+	fmt.Printf("Found %v commits. \n", len(commits))
```

Comment:
Can you print the commits here as well, when I tried using the backport tool I realized that would be pretty useful. So something like:

```
Found 2 commit(s).
e45ab655 Fix reverse tunnel dialing for Windows Desktops
730352c3 omit invalid aws tags in rds autodiscovery
```